mrmustard/physics/gaussian_integrals.py

```diff
@@ -370,11 +374,11 @@ def true_branch_complex_gaussian_integral_1(m, M, bM, det_M, c, D, R, bR):
     inv_M = math.inv(M)
     M_bM = math.solve(M, bM)
 
-    c_factor = math.sqrt(math.cast((-1) ** m / det_M, "complex128")) * math.exp(
-        -0.5 * math.sum(bM * M_bM, axis=-1),
+    log_c_factor = math.log(math.sqrt(math.cast((-1) ** m / det_M, "complex128"))) + (
+        -0.5 * math.sum(bM * M_bM, axis=-1)
     )
-    c_reshaped = math.reshape(c_factor, batch_shape + (1,) * (len(c.shape[batch_dim:])))
-    c_post = c * c_reshaped
+    log_c_reshaped = math.reshape(log_c_factor, batch_shape + (1,) * (len(log_c.shape[batch_dim:])))
+    c_post = math.exp(log_c + log_c_reshaped)
```

Comment:
I'm somewhat surprised that this is necessary. Can you point to tests that fail if we _don't_ do this? That is, are there tests that were not simply "updated" but also used as part of test-driven development?

Reply:
The
```python
    @pytest.mark.parametrize("alpha", [10 + 10j, 18 + 18j, 25 + 25j])
    @pytest.mark.parametrize("r", [0, 1, 2, 3])
    def test_probability(self, alpha, r):
        """Tests that highly displaced squeezed states are properly normalized."""
        state = DisplacedSqueezed(mode=0, alpha=alpha, r=r)
        assert math.allclose(state.probability, 1)
```
got it. You don't even need a `DisplacedSqueezed` state to do this though, right? It's the large displacement that is causing an issue, as it leads to a very small `c` (as it's `exp(-|alpha|^2 / 2)`)? That is, the issue appears to be about numbers that are too small to be represented by `complex128`, not necessarily about the product of a large and a small number.

Reply:
Somewhat has me wondering if there should be warnings about creating states that will have such issues. Already, the probability is `nan` for a state like `Coherent(mode=0, alpha=26.64)`.

Reply:
I just moved the test to change from being a test of DisplacedSqueezed to a test of Coherent.

I agree this won't fix everything. There will still be a limit where things break, as you've found. Not sure what best solution is.

On this branch, `Coherent(mode=0, alpha=26.64)` works. Things start to break down as you go higher than `Coherent(0, 38.2).probability`. By `Coherent(0, 38.7).probability` I get a warning about an invalid encountered in log.

Reply:
In the future, we should probably work to raise that `.probability` is not possible to be calculated for such states at the time they're created instead of only when `.probability` is called, and potentially consider using A,b,log(c) instead of A,b,c throughout the codebase, for for now I think the changes hare are good.